src/util/integer_gmp_imp.cpp

```diff
@@ -340,6 +340,21 @@ bool Integer::isNegativeOne() const
 }
 
 Integer Integer::pow(unsigned long int exp) const
+{
+  // check if exp is within the uint32_t range
+  uint32_t low = static_cast<uint32_t>(exp); 
+  uint32_t high = static_cast<uint32_t>(exp << 32);
+  if (low == exp) {
+    // if it is, safely call the gmp pow function
+    return powHelper(exp);
+  } else {
+      // if exp is bigger than max uint32_t, use the following identity:
+      // a^(x + 2^32*y) = a^x * (a^(2^32))^y
+      return powHelper(low) * ((Integer(static_cast<uint32_t>(-1))) * (*this)).powHelper(high);
```

Comment:
I don't think this is correct: it should have three calls to `powHelper()`, but only has two. Please also add one or two unit tests that actually exercise power-by-more-than-2^32 :)

Reply:
fixed.
However, regarding the tests — they take too much time to be added as unit tests I believe. I tried also using max-unsigned-int + 1 and it was too slow.